onnxscript/rewriter/ort_fusions/sdpa.py

```diff
@@ -57,34 +58,51 @@ def check(self, op, query, key_transposed, value, mask, query_scale, key_scale,
 
         if self._pre_scale:
             # Check if query_scale and key_scale are scalars == sqrt(expected_scaling_factor)
+            # If they are scalars but != sqrt(expected_scaling_factor), a custom scale is being used.
             sqrt_scaling_factor = math.sqrt(expected_scaling_factor)
-            if not _ir_utils.is_singleton_value(query_scale, sqrt_scaling_factor, rtol=1e-3):
+            # Calculate the scaling factor for query
+            if (query_scale_value := _ir_utils.get_singleton_value(query_scale)) is None:
                 return check_result.fail(
-                    "Query scale is not a scalar or does not match the expected scaling factor.",
+                    "Query scale is not a scalar.",
                     query_scale,
                 )
-            if not _ir_utils.is_singleton_value(key_scale, sqrt_scaling_factor, rtol=1e-3):
+            if not math.isclose(query_scale_value, sqrt_scaling_factor, rel_tol=1e-3):
+                self._scale = query_scale_value * query_scale_value
+            else:
+                self._scale = expected_scaling_factor
```

Comment:
Should we be setting this to `None`, if we want to indicate it is the default-value? If we are not doing that, not sure we even need this if-then-else, we can just use like 70 always?

Reply:
Setting it to `None` makes more sense. Pass scale(=None) to SDPA for these cases, so the SDPA implementation knows that default scale value is to be used). Obviously, we might need to adjust the semantics once we have the attention/SDPA to test against, but for now this should be good.